### DISCUSSION TITLE
Selectively decrease PolygonOffset of few items in data array of layer

### DISCUSSION THREAD

**Discussion by Anand2489** [visgl/deck.gl#6875] [Q&A] [answered]
Depending on some user interaction or condition, I have to increase z-index of few items from `data` prop array. 

`getPolygonOffset` applies depth offset to entire layer. Can it be done selectively within the same layer.

One way for doing it is filtering out items to which we want to increase z-index & then create 2 layers, one with lower value of PolygonOffset(to bring it closer to the camera) & other without it! I'm not sure how performant it would be for large datasets.

Any better approach suggested by the community ?

**Comment by Pessimistress** [comment, marked as answer]
> `getPolygonOffset` applies depth offset to entire layer

That is correct. All layers use instanced rendering, which means all objects are rendered with a single draw call. It's not possible to set different GL parameters without splitting them into separate layers.

**Comment by Anand2489** [comment]
@Pessimistress Thanks for your reply :)

Do you have any better approach to full my use case, that is, I have rendered large set of polylines using DeckGL's PathLayer. Now depending on user selection of some of these polylines, I have to bring them to the front so that these selected lines are always visible even if user zooms out..
